libcaf_core/test/make_config_value_field.cpp

```diff
@@ -0,0 +1,261 @@
+    CAF_CHECK_EQUAL(foo_field.get(x), config_value(42));
+    CAF_CHECK(!foo_field.type_check(config_value(1.)));
+    CAF_CHECK(foo_field.type_check(config_value(-1)));
+    foo_field.set(x, config_value(-1));
+    CAF_CHECK_EQUAL(foo_field.get(x), config_value(-1));
+    CAF_CHECK(!foo_field.valid(x));
+    string_view input = "123";
+    string_parser_state ps{input.begin(), input.end()};
+    foo_field.parse_cli(ps, x);
+    CAF_CHECK_EQUAL(ps.code, pec::success);
+    CAF_CHECK_EQUAL(foo_field.get(x), config_value(123));
+  }
+
+  template <class T>
+  expected<T> read(std::vector<std::string> args) {
+    settings cfg;
+    auto res = opts.parse(cfg, args);
+    if (res.first != pec::success)
+      return make_error(res.first, *res.second);
+    auto x = get_if<T>(&cfg, "value");
+    if (x == none)
+      return sec::invalid_argument;
+    return *x;
+  };
+};
+
+} // namespace
+
+namespace caf {
+
+template <>
+struct config_value_access<foobar> : config_value_object_access<foobar_trait> {
+};
+template <>
+struct config_value_access<foobar_foobar>
+  : config_value_object_access<foobar_foobar_trait> {};
+
+} // namespace caf
+
+CAF_TEST_FIXTURE_SCOPE(make_config_value_field_tests, fixture)
+
+CAF_TEST(construction from pointer to member) {
+  make_config_value_field("foo", &foobar::foo);
+  make_config_value_field("foo", &foobar::foo, none);
+  make_config_value_field("foo", &foobar::foo, none, nullptr);
+  make_config_value_field("foo", &foobar::foo, 42);
+  make_config_value_field("foo", &foobar::foo, 42, nullptr);
+  make_config_value_field("foo", &foobar::foo, 42, foo_valid);
+  make_config_value_field("foo", &foobar::foo, 42,
+                          [](const int& x) { return x != 0; });
+}
+
+CAF_TEST(pointer to member access) {
+  auto foo_field = make_config_value_field("foo", &foobar::foo, 42, foo_valid);
+  test_foo_field(foo_field);
+}
+
+CAF_TEST(construction from getter and setter) {
+  auto get_foo_lambda = [](const foobar& x) { return x.foo; };
+  auto set_foo_lambda = [](foobar& x, int value) { x.foo = value; };
+  make_config_value_field("foo", get_foo, set_foo);
+  make_config_value_field("foo", get_foo_fun, set_foo);
+  make_config_value_field("foo", get_foo_fun, set_foo_fun);
+  make_config_value_field("foo", get_foo_lambda, set_foo_lambda);
+}
+
+CAF_TEST(getter and setter access) {
+  auto foo_field = make_config_value_field("foo", get_foo, set_foo, 42,
+                                           foo_valid);
+  test_foo_field(foo_field);
+}
+
+CAF_TEST(oject access from dictionary - foobar) {
+  settings x;
+  put(x, "my-value.bar", "hello");
+  CAF_MESSAGE("without foo member");
+  {
+    CAF_REQUIRE(holds_alternative<foobar>(x, "my-value"));
+    CAF_REQUIRE(get_if<foobar>(&x, "my-value") != caf::none);
+    auto fb = get<foobar>(x, "my-value");
+    CAF_CHECK_EQUAL(fb.foo, 123);
+    CAF_CHECK_EQUAL(fb.bar, "hello");
+    CAF_MESSAGE("with foo member");
+  }
+  put(x, "my-value.foo", 42);
```

Comment:
Not that important, but the `CAF_MESSAGE` comments on the put outside of the block. Would make sense to group them together.